src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
ClientTemplate is a trait of course.

Yes, bodies would be value objects. Drop what methods?

You can take a look at the body branch. Currently the interface accepts a plain string body, a StreamInterface or a Body object. Body objects must be string castable. Also, in some cases (for example in case of Files) the request headers must be altered, so I am thinking about to add an alterRequest method to the Body interface which allows to add required headers.